src/UCTSearch.cpp

```diff
@@ -413,7 +413,7 @@ int UCTSearch::get_best_move(passflag_t passflag) {
     assert(first_child != nullptr);
 
     auto bestmove = first_child->get_move();
-    auto bestscore = first_child->get_eval(color);
+    auto bestscore = first_child->get_pure_eval(color);
```

Comment:
No guarantee yet the first child has visits?

Reply:
Does that guarantee exist with `get_eval` though? Maybe make it `0.5f` or `1.0f` with no visits?

Reply:
Your point is right, but get_eval works by coincidence due to virtual loss and so won't ever crash here, which the new code would.

Reply:
I've changed it, but I don't see how `first_child` is guaranteed to have a virtual loss.

Reply:
Hmm, it's not (it's actually guaranteed not to in get_best_move). But if it isn't, then changing the calls from get_eval to get_pure_eval is a no-op obviously.

Maybe it's good to check what exactly happens with p = 1 and see if we get the edge cases right? And is specifying p = 0 possible?